docs/collisions.md

```diff
@@ -0,0 +1,571 @@
+
+To make this possible, a new property `downloadName` would be added to the data models of files and folders each. The intention is that it is the downloadName that would be used when mapping files and folders from Permanent’s file system to other file systems.
+
+
+## Definitions
+
+
+###  Conflict/Name Collision
+
+
+There is a name collision or conflict **IF more than one file and or folder in the same directory** has **the same name and extension when applicable. Summarily;**
+
+
+
+* More than one file with an exact name and extension is a conflict
+* More than one folder with an exact name including extension
+* More than one folder and file with an exact name including or excluding extension
+
+
+###  Siblings
+
+Files and folders that are inside the same directory would be referred to as **siblings.**
+
+
+## How does `downloadName` work
+
+The value of the **_downloadName_** field is obtained from the **_displayName._** This implies `downloadName = displayName + file_extension` when there is no naming conflict.
+
+In the event of a conflict,
+
+For example if a folder has five 5 with the same name say **_A.txt _** the five files would have downloadName ‘s as `A, A (1).txt, A (2).txt, A (3).txt, A (4).txt`
+
+
+### Algorithm
+
+
+* **1: IF SIBLINGS :** check for conflicts
+* **2: IF NO CONFLICT :** The `downloadName = displayName + fileExtension` or simply  `downloadName = displayName` in the case of folder.
+* **3: IF CONFLICT:** The` downloadName = displayName (fileId) + fileExtension`
+
+**Notes:**
+
+
+* A proposal was made to use the actual filesystem convention for handling name collisions which is  “space, parentheses open, number, parentheses close”  for example (2232)
+* A proposal was made to use the **recordId or FolderId **within the parenthesis instead of just natural incrementation. This would make downloadName collisions impossible and checking for available names unnecessary.
+* A proposal was made to not even store **downloadName **in the database since all the parameters that make it up are unique and already available. Instead they would be calculated as needed. [There would be a good case to store it though to make operations faster]
+* Name collisions should be checked across record and folder tables
+* [Extensions in folder names matter](https://apple.stackexchange.com/questions/123001/renamed-folder-becomes-a-file-with-an-extension) and should be handled :
+* Case sensitivity is taken into account. `Myfile` and `myFile` are different for example. That is useful to [ensure complete syncs](https://rclone.org/overview/#case-insensitive).
+
+### Reserved/unsupported characters
```

Comment:
@slifty commits stashed and this section added!

Reply:
@Fenn-CS Guessing you meant "squashed" not "stashed"?

Reply:
@kfogel Yeah thanks, I mean squashed. Git and its terms 😀🤦🏽‍♂️